### DISCUSSION TITLE
Capstone Data Visualization : Mengubah interval/breaks Date

### DISCUSSION THREAD

**Discussion by daryraa** [teamalgoritma/community#262] [Q&A] [answered]
Saya ingin interval datanya berubah /dikasih breaks supaya lebih rapih itu bagaimana ya
misal saya cuma ingin kasih rentang nilai pertahun saja, jadi yang perhari tidak ditampilkan
![image](https://user-images.githubusercontent.com/78478403/183418985-5a3f26c1-c4aa-4e23-b78f-9682eac9803e.png)

sales_profit<- sales_clean %>% 
      group_by(Date) %>% 
      summarise(sum_profit = sum(Profit)) %>% 
      ungroup() 
    
    sales_profit <- sales_profit %>% 
      mutate(
        label2 = glue(
        "Year: {Date}
        Profit: {comma(round(sum_profit, 2))}"
      ))
    label_rupiah <- label_dollar(prefix = "$ ")
    plot1 <- ggplot(sales_profit, aes(x= Date, y= sum_profit))+
      geom_line(col="blue") +
      geom_point(aes(text=label2), col="black") +
      labs(
        title = "Total Profit",
        x = "Date",
        y = NULL
      ) +
      scale_x_date(date_breaks = "1 years",date_labels = "%Y %b") +
      scale_y_continuous(labels = label_rupiah)+
      theme_minimal()
    
    ggplotly(plot1, tooltip = "text")

**Comment by IrfanChairurrachman** [comment, marked as answer]
Halo @daryraa . Tujuan visualisasi adalah menampilkan total profit tahunan hasil agregasi dari data harian ya? Jika iya, maka permasalahan ini terjadi pada agregasi datanya. Jadi pada kode agregasi yang dilampirkan di atas, bahwa di groupingnya menggunakan kolom `Date`, dan contoh bentuk dari kolom `Date` adalah `2011 Jan`, `2012 Jan` dll. Apakah benar bahwa kolom `Date` datanya berbentuk demikian?

Jika benar, maka kami menyarankan untuk mengubah dulu bentuk kolom `Date` menjadi tipe data `datetime`, dalam kasus ini dapat digunakan library `lubridate` dengan fungsi `ym()` seperti ini:
`sales_clean$Date <- ym(sales_clean$Date)`

Setelah mengubah tipe data, silahkan ekstrak informasi tahun ke dalam kolom baru dari kolom `Date`, misalnya informasi tahun kita letakkan di dalam kolom `year`:
`sales_clean$year <- year(sales_clean$Date)`.

Dapat dilihat bahwa kolom `year` merepresentasikan tahun dari setiap baris yang ada. Selanjutnya baru dapat dilakukan agregasi data untuk menghitung total profit pertahun menggunakan group_by(year) dan dimasukkan ke dalam `sales_profit`, pastikan data `sales_profit` adalah hasil agregasi jumlah profit pertahun. Kemudian dapat dilakukan visualisasi lagi sehingga yang ditampilkan hanyalah data total profit pertahunnya, bukan untuk masing-masing hari.

Sekian, semoga membantu. Mohon konfirmasi setelah mencoba cara ini ya dengan reply diskusi ini. Terima kasih

**Reply by daryraa** [reply]
baik mas terima kasih, akan saya coba

**Reply by IrfanChairurrachman** [reply]
Untuk @daryraa , masukan kami terhadap Capstone bapak, agar visualiasi lebih rapi, kami menyarankan untuk menampilkan data perbulan saja alih-alih menampilkan data perhari, adapun tahapan yang digunakan yaitu membuat kolom baru untuk informasi Year-Month, melakukan agregasi data, kemudian mengubah tipe data Year_Month kembali menjadi `Date` agar dapat digunakan oleh fungsi `scale_x_date()`. 

```
# Membuat kolom baru dengan informasi Year-Month
sales_clean$Year_month <- format(as.Date(sales_clean$Date), '%Y-%m')

sales_profit_Ym <- sales_clean %>% 
      group_by(Year_month) %>% 
      summarise(sum_profit = sum(Profit)) %>% 
      ungroup() %>% 
      mutate(Year_month = ym(Year_month)) # Mengubah tipe data dari chr menjadi Date agar bisa diolah ggplotly
```
Jika data perbulan, maka tampilan akan seperti ini

![image](https://user-images.githubusercontent.com/56690035/183828058-d2bf14dd-34c6-4de7-9d6b-924619771cb6.png)
